sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -328,7 +329,7 @@ object PartitioningUtils {
     } else {
       // TODO: Selective case sensitivity.
       val distinctPartColNames =
-        pathsWithPartitionValues.map(_._2.columnNames.map(_.toLowerCase())).distinct
+        pathsWithPartitionValues.map(_._2.columnNames.map(_.toLowerCase(Locale.ROOT))).distinct
```

Comment:
I think this might cause a similar problem with https://github.com/apache/spark/pull/17527/files#r110317272.